gillespy2/solvers/cpp/c_solver.py

```diff
@@ -331,6 +331,8 @@ def _update_resume_data(self, resume: Results, simulation_data: "list[dict[str,
         for entry_name, entry_data in simulation_data[-1].items():
             # The results of the current simulation is treated as an "extension" of the resume data.
             # As such, the new simulation output is formed by joining the two end to end.
+            if entry_name != "time":
+                entry_data = entry_data[:-1]
```

Comment:
I did some testing with pause/resume models, and it showed that there was consistently a duplicate entry from the end time of the initial run to the "next step" of the resume run.  This fix shaves the last data point off so that things line up, but I have proposed an alternate fix in #643 which eliminates the duplicate entries.